arangod/V8Server/v8-actions.cpp

```diff
@@ -436,6 +436,62 @@ v8::Handle<v8::Object> TRI_RequestCppToV8(v8::Isolate* isolate,
   // intentional copy, as we will modify the headers later
   auto headers = request->headers();
 
+  switch (request->acceptEncoding()) {
+  case EncodingType::UNSET:
+    headers.insert(std::pair<std::string, std::string>(StaticStrings::AcceptEncoding, StaticStrings::EncodingIdentity));
+    break;
+  case EncodingType::DEFLATE:
+    headers.insert(std::pair<std::string, std::string>(StaticStrings::AcceptEncoding, StaticStrings::EncodingDeflate));
+    break;
+  }
+  std::string const& acceptPlain = request->contentTypeResponsePlain();
+
+  if (!acceptPlain.empty()) {
+    headers.insert(std::pair<std::string, std::string>(StaticStrings::Accept, acceptPlain));
+  } else {
+    switch(request->contentTypeResponse()) {
+    case ContentType::UNSET:
+    case ContentType::CUSTOM:  // use Content-Type from _headers
```

Comment:
Mmh, this `switch` statement here has a different indentation than the one added a few lines below for the content type.

Reply:
yes, its in an else-case. `'image/webp,text/html,application/x-html,*/*;q=0.8'`would become `text/html` here without the if and the storing of this exact header previously.